### PR TITLE
Clear out all -Wunused warnings from gcc 7.2

### PR DESCRIPTION
**Pull request by grahamlopez**
If something important fell victim, let's figure out why it was showing up "unused."

### CODE REVIEW

File: src/Drivers/check_spo.cpp [hunk 1]
```diff
@@ -73,9 +69,6 @@ int main(int argc, char **argv)
     case 'i': // number of MC steps
       nsteps = atoi(optarg);
       break;
-    case 's': // random seed
```
Comment (markdewing) [inline src/Drivers/check_spo.cpp]: This should probably stay, and get hooked up to the random generator.  It looks like the code generates random electron positions.   Any code that uses random numbers should have a way to specify the seed.

File: src/Drivers/check_wfc.cpp [hunk 2]
```diff
@@ -52,9 +52,7 @@ void print_help()
   cout << "      one of: J1, J2, J3.            default: J2"            << '\n';
   cout << "  -g  set the 3D tiling.             default: 1 1 1"         << '\n';
   cout << "  -h  print help and exit"                                   << '\n';
-  cout << "  -i  number of Monte Carlo steps.   default: 100"           << '\n';
   cout << "  -r  set the Rmax.                  default: 1.7"           << '\n';
-  cout << "  -s  set the random seed.           default: 11"            << '\n';
```
Comment (markdewing) [inline src/Drivers/check_wfc.cpp]: This seed option should probably stay, and get hooked up.   It appears this code uses the same implicit seed all the time (unlike check_spo, which generates the seed from the time).

File: src/Drivers/miniqmc.cpp [hunk 3]
```diff
@@ -158,6 +155,7 @@ int main(int argc, char **argv)
       break;
     case 'g': // tiling1 tiling2 tiling3
       sscanf(optarg, "%d %d %d", &na, &nb, &nc);
+      optind += 2;
```
Comment (markdewing) [inline src/Drivers/miniqmc.cpp]: The expected form for na,nb,nc separated by spaces pretty much requires them to be enclosed by quotes on the command line (miniqmc -g "2 2 4").  If enclosed in quotes, then they form a single argument.   I don't think incrementing optind is correct here.

**Comment by grahamlopez** [comment]
@markdewing I addressed your comments, and went ahead and made all of the cli input processing the same across the drivers.

**Comment by markdewing** [comment]
@ye-luo Are you okay with these changes?

**Comment by ye-luo** [comment]
These changes are OK. I just have a feeling that something not used currently may not be useless but we can add them back if needed later.

**Pull request merged**